provider/datagen/src/provider.rs

```diff
@@ -0,0 +1,479 @@
+                        )
+                        .as_slice(),
+                    ),
+                    (
+                        "Lao_codepoints_exclusive_model4_heavy/weights.json",
+                        include_bytes!(
+                            "../tests/data/lstm/Lao_codepoints_exclusive_model4_heavy/weights.json"
+                        )
+                        .as_slice(),
+                    ),
+                    (
+                        "Burmese_codepoints_exclusive_model4_heavy/weights.json",
+                        include_bytes!(
+                            "../tests/data/lstm/Burmese_codepoints_exclusive_model4_heavy/weights.json"
+                        )
+                        .as_slice(),
+                    ),
+                    (
+                        "Thai_codepoints_exclusive_model4_heavy/weights.json",
+                        include_bytes!(
+                            "../tests/data/lstm/Thai_codepoints_exclusive_model4_heavy/weights.json"
+                        )
+                        .as_slice(),
+                    ),
+                    (
+                        "Thai_graphclust_model4_heavy/weights.json",
+                        include_bytes!("../tests/data/lstm/Thai_graphclust_model4_heavy/weights.json")
+                            .as_slice(),
+                    ),
+                ]
+                .into_iter()
+                .collect(),
+            )))),
+            ..DatagenProvider::default().source
+        }
+    }
+}
+
+#[cfg(feature = "legacy_api")]
+impl SourceData {
+    /// See [`DatagenProvider::LATEST_TESTED_CLDR_TAG`]
+    pub const LATEST_TESTED_CLDR_TAG: &'static str = DatagenProvider::LATEST_TESTED_CLDR_TAG;
+
+    /// See [`DatagenProvider::LATEST_TESTED_ICUEXPORT_TAG`]
+    pub const LATEST_TESTED_ICUEXPORT_TAG: &'static str =
+        DatagenProvider::LATEST_TESTED_ICUEXPORT_TAG;
+
+    #[cfg(feature = "networking")]
+    /// See [`DatagenProvider::latest_tested`]
+    pub fn latest_tested() -> Self {
+        DatagenProvider::latest_tested().source
+    }
+
+    /// See [`DatagenProvider::with_cldr`]
+    pub fn with_cldr(
+        self,
+        root: PathBuf,
+        _use_default_here: crate::CldrLocaleSubset,
+    ) -> Result<Self, DataError> {
+        Ok(DatagenProvider { source: self }.with_cldr(root)?.source)
+    }
+
+    /// See [`DatagenProvider::with_icuexport`]
+    pub fn with_icuexport(self, root: PathBuf) -> Result<Self, DataError> {
+        Ok(DatagenProvider { source: self }
+            .with_icuexport(root)?
+            .source)
+    }
+
+    #[cfg(feature = "networking")]
+    /// See [`DatagenProvider::with_cldr_for_tag`]
+    pub fn with_cldr_for_tag(
+        self,
+        tag: &str,
+        _use_default_here: crate::CldrLocaleSubset,
+    ) -> Result<Self, DataError> {
+        Ok(DatagenProvider { source: self }
+            .with_cldr_for_tag(tag)
+            .source)
```

Comment:
Suggestion (optional): seems ok to have the real functions on DatagenProvider but it might make more sense to keep them on SourceData and have DatagenProvider call the SourceData functions instead of the other way around since SourceData is the stateful thing